From 7d9035c6d6ddf336478b61d5f5796442990c059a Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Tue, 28 Jan 2025 15:31:53 +0100
Subject: [PATCH] clk: stm32: fix condition for calling usleep_range in reset
 ops

The usleep_range function should be called only when data->reset_us
is non-zero.

Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: Ic16e39688d603d422b6514abdb18d9ddd416735c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/438037
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 drivers/clk/stm32/reset-stm32.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/clk/stm32/reset-stm32.c
+++ b/drivers/clk/stm32/reset-stm32.c
@@ -145,7 +145,7 @@ static int stm32_reset(struct reset_cont
 	if (ret)
 		return ret;
 
-	if (!data->reset_us)
+	if (data->reset_us)
 		usleep_range(data->reset_us, data->reset_us * 2);
 
 	ret = stm32_reset_deassert(rcdev, id);
